pilot/pkg/proxy/envoy/v3/model.go

```diff
@@ -14,11 +14,36 @@
 
 package v3
 
-import "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+import (
+	resourcev2 "github.com/envoyproxy/go-control-plane/pkg/resource/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+)
 
 const (
 	ClusterType  = resource.ClusterType
 	EndpointType = resource.EndpointType
 	ListenerType = resource.ListenerType
 	RouteType    = resource.RouteType
 )
+
+var (
+	ListenerShortType = "LDS"
+	RouteShortType    = "RDS"
+	EndpointShortType = "EDS"
+	ClusterShortType  = "CDS"
+)
+
+func GetShortType(typeURL string) string {
```

Comment:
I missed this - why do we need the 'short types' ?

Reply:
comment explaining this is on the TypeUrl, probably should be here though. Basically, we don't want to key by TypeUrl, because we should have the same key for v2.Cluster and v3.Cluster. So this transforms typeurl -> key. We could just translate v2 types to v3, I don't really care either way - its all internal anyways